src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/StaticAssetsIntegrationTest.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.Extensions.CommandLineUtils;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
+{
+    public class StaticAssetsIntegrationTest : MSBuildIntegrationTestBase, IClassFixture<BuildServerTestFixture>, IClassFixture<PackageTestProjectsFixture>
+    {
+        public StaticAssetsIntegrationTest(
+            BuildServerTestFixture buildServer,
+            PackageTestProjectsFixture packageTestProjects,
+            ITestOutputHelper output)
+            : base(buildServer)
+        {
+            UseLocalPackageCache = true;
+            packageTestProjects.Pack(output);
+        }
+
+        [Fact]
+        [InitializeTestProject("AppWithPackageAndP2PReference")]
+        public async Task Build_GeneratesAspNetCoreStaticAssetsManifest_Success_CreatesManifest()
+        {
+            // For some reason when using a custom package cache the imports won't get added on
+            // the initial restore, so we restore the packages ourselves.
+            await DotnetMSBuild("Restore");
```

Comment:
@pranavkm

Reply:
This didn't cause any issue in my prototype, I have pending to investigate the full extent of the issue here. Successive restores are fine and I don't recall seeing this when I was not using a local cache. But I'll dig in a bit later.

Reply:
It doesn't affect rebuilds either (have a test for that)

Reply:
Can we have this time out really quickly? This should only ever be going to disk which should have a reasonable restore time.

Reply:
This does go to disk because we are using a custom nuget packages folder in order not to pollute the global namespace.